drivers/filesystems/fastfat/misc.c

```diff
@@ -416,23 +416,43 @@ VfatLockUserBuffer(
 BOOLEAN
 VfatCheckForDismount(
     IN PDEVICE_EXTENSION DeviceExt,
-    IN BOOLEAN Create)
+    IN BOOLEAN Force)
 {
     KIRQL OldIrql;
+    ULONG UnCleanCount = 3; // FIXME: or 2 depending on whether we are in IRP_MJ_CREATE; however I think we still leak like 1 or 2 extra refs there...
```

Comment:
What's the point of this?

Reply:
Excerpt from the commit message:

> NOTE: In their *CheckForDismount() function, our 3rd-party FS drivers
as well as MS' fastfat, perform a comparison check of the current VCB's
VPB ReferenceCount with some sort of "dangling"/"residual" open count.
It seems to be related to the fact that the volume root directory as
well as auxiliary data stream(s) are still opened, and only these are
allowed to be opened at that moment. After analysis it appears that for
the ReactOS' fastfat, this number is equal to "3". But this should be
investigated more closely...

Reply:
Any ideas / advice about this are welcome.